design/app-to-app.md

```diff
@@ -0,0 +1,80 @@
+App to App Interactions
+===========
+
+This document describes an extension to the SDK to allow MRE apps to receive content (e.g., photos) from the MRE clients. Content may be generated by the client itself, or, in future iterations not yet described here, may arrive from other apps with coordination from a particular client.
+
+API Design
+-----------
+This design introduces the concept of a `SharingMessage`, a base class of content-specific messages, such as `PhotoSharingMessage`. We provide an outline for the implementation of `PhotoSharingMessage` only, but expect to add more sharing capabilities over time.
+
+`SharingMessage` is itself relatively straightforward, but contains enough information to allow the recipient (callback listener) to cast it to the appropriate type (e.g., `PhotoSharingMessage`).
+
+```ts
+class SharingMessage {
+    messageType: Readonly<string>;
+    sessionId: Readonly<string>;
+    userId: Readonly<Guid>;
+}
+
+class PhotoSharingMessage extends SharingMessage {
+    photoType: Readonly<string>; // Supported values: "jpg" and "png"
```

Comment:
Is a string the right way to go here? How will the difference between "jpg" and "jpeg" be handled?

Reply:
Good question: As written, "jpeg" wouldn't be accepted. Open to suggestions. Could add it to the list, note explicitly that the client implementation should use "jpg" rather than "jpeg", or keep it as is.

Reply:
TODO : change to MIME Type. And note that any other types (e.g., heic) would be on the mre host to convert.